ibis/backends/dask/execution/arrays.py

```diff
@@ -38,12 +38,54 @@ def execute_array_column(op, cols, **kwargs):
     )
 
 
+@execute_node.register(ops.ArrayConcat, dd.Series, dd.Series)
+def execute_array_concat_series(op, left, right, **kwargs):
+    df = dd.concat([left, right], axis=1)
+    return df.apply(lambda row: np.concatenate(row), axis=1, meta=left._meta,)
+
+
+@execute_node.register(ops.ArrayConcat, dd.Series, np.ndarray)
+@execute_node.register(ops.ArrayConcat, np.ndarray, dd.Series)
+def execute_array_concat_mixed(op, left, right, **kwargs):
+    if isinstance(left, dd.Series):
+        # "Iterate" over `left` Series, concatenating each array/element
+        # in the Series with `right`, which is a single ndarray.
+        return left.apply(
+            lambda arr: np.concatenate([arr, right]), meta=left._meta,
+        )
+    elif isinstance(right, dd.Series):
+        return right.apply(
+            lambda arr: np.concatenate([left, arr]), meta=right._meta,
+        )
```

Comment:
Nitpick - but I think this would be a bit clearer if you `if/else` on the types an only have one `return/apply`. Like this, but with better names:

```python
def execute_array_concat_mixed(op, left, right, **kwargs):
    s, npa = (left, right) if isinstance(left, dd.Series) else (right, left)
    
    return s.apply(
        lambda a: np.concatenate([a, npa]), meta=s._meta
    )
```